src/Appointment/Request.js

```diff
@@ -0,0 +1,328 @@
+import React, {useState} from 'react';
+import Card from 'react-bootstrap/Card';
+import Button from 'react-bootstrap/Button';
+import './Request.css';
+
+class Request extends React.Component {
+
+    constructor() {
+
+      
+    super();
+
+    this.state = {
+
+      input: {},
+
+      errors: {}
+
+    };
+
+     
+
+    this.handleChange = this.handleChange.bind(this);
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+
+  }
+
+    
+
+  handleChange(event) {
+
+    let input = this.state.input;
+
+    input[event.target.name] = event.target.value;
+
+  
+
+    this.setState({
+
+      input
+
+    });
+
+  }
+
+
+     
+
+  handleSubmit(event) {
+
+    event.preventDefault();
+
+  
+
+    if(this.validate()){
+
+        console.log(this.state);
+
+  
+
+        let input = {};
+
+        input["name"] = "";
+
+        input["email"] = "";
+
+        input["phone"] = "";
+
+        input["date"] = "";
+
+
+        // input["address"] = "";
+
+        this.setState({input:input});
+
+  
+
+        alert('Appointment request is submited');
+
+    }
+
+  }
+
+  validate(){
+
+      let input = this.state.input;
+
+      let errors = {};
+
+      let isValid = true;
+
+  
+
+      if (!input["name"]) {
+
+        isValid = false;
+
+        errors["name"] = "Please enter your name.";
+
+      }
+
+  
+
+      if (!input["email"]) {
+
+        isValid = false;
+
+        errors["email"] = "Please enter your email Address.";
+
+      }
+
+  
+
+      if (typeof input["email"] !== "undefined") {
```

Comment:
I see we have a lot of if logic here. Probably should use a util function to check all of this out. If you want we can take a look at this together on weekend. over a zoom call with Vaidehi as well.